### DISCUSSION TITLE
How do I create a Chatbot that retrieves specific documents that a user requests for?

### DISCUSSION THREAD

**Discussion by RobertoGongora** [LlmLaraHub/laralamma#21] [Q&A]
I have a collection of misc documents that can span from legal requirements to event brochures for meetups. I want to provide my users with a chatbot interface where they can request specific documents in natural language, like: "I want to download the event brochure for the September meetup" -- and get either a summary with a download URL or a list of URLs if there are multiple results.

The download URL is stored as part of each document's metadata in the collection.

**Comment by alnutile** [comment]
That is a good question and I will need to add it as a feature request since it is missing in other areas as well (for example I send CMS updates to a collection and it includes web url of the page which would be great to include in the output) 
Basically the part of the code that does the work to query and find related content has to include that "file_path" or url info in the response.

For example https://github.com/LlmLaraHub/larallama/blob/main/Modules/LlmDriver/app/Functions/SummarizeCollection.php#L35

Any function that searches the database like above or https://github.com/LlmLaraHub/larallama/blob/main/Modules/LlmDriver/app/Functions/SearchAndSummarize.php#L78

Just needs to include more than just summary in the results. File Path, or what ever makes sense for that document type.
the Document model has some fields but it also has meta_data https://github.com/LlmLaraHub/larallama/blob/main/app/Models/Document.php#L38

So the Type of document can then be smart enough to get the right meta_data URL, Public links etc.
